torch/_tensor_docs.py

```diff
@@ -2162,16 +2166,22 @@ def callable(a, b) -> number
 for ``dimension != dim`` and by the corresponding value in :attr:`index` for
 ``dimension = dim``.
 
+If :attr:`other` is a tensor, then :attr:`self` and :attr:`other` should have same number
+of dimensions. Otherwise, :attr:`other` must be a scalar that have the same dtype as :attr:`self`.
+:attr:`index` should also have the same number of dimensions as :attr:`self` and :attr:`other`.
+It is also required that ``index.size(d) == other.size(d) == self.size(d)`` for all dimensions ``d != dim``.
```

Comment:
this description also seems wrong -- are there really no constraints when `d == dim` ?